src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerArgumentCollection.java

```diff
@@ -133,7 +103,7 @@ public ReadThreadingAssembler createReadThreadingAssembler() {
 
     @Advanced
     @Argument(fullName = SMITH_WATERMAN_LONG_NAME, doc = "Which Smith-Waterman implementation to use, generally FASTEST_AVAILABLE is the right choice", optional = true)
-    public SmithWatermanAligner.Implementation smithWatermanImplementation = SmithWatermanAligner.Implementation.JAVA;
+    public SmithWatermanAligner.Implementation smithWatermanImplementation = SmithWatermanAligner.Implementation.FASTEST_AVAILABLE;
```

Comment:
@davidbenjamin Did you intend for this PR to make substantive changes to default values for arguments? You present it as a pure refactoring branch, but it looks like there are changes like this embedded in it. We deliberately have not yet turned the native SmithWaterman on by default due to our current inability to continuously test it.

Reply:
@droazen That's an accident -- I was just curious about its effect on the M2 integration test runtime and forgot to revert.  I intend everything to be refactoring.  This does include simplifying the assembly argument collections, which is not strictly necessary for the purposes of this PR.  I could strip out things like that if you would prefer.

Reply:
Ah ok, thanks for the clarification! Can you do a pass to check that additional changes like this didn't slip in (ie., changes to argument defaults)?

Reply:
Absolutely!

Reply:
. . . That was the only one, fortunately.